<launch>
<arg name="robot"/>
<!-- autostart: true to start the PPLP at startup -->
<arg name="autostart" default="false"/>
<!-- eventgui: true to create a GUI that allows to start the PPLP -->
<arg name="eventgui"  default="false"/>
<!-- marker: true to create a node converting the published PPL into a Vizualisation marker -->
<arg name="marker"    default="false"/>

<group ns="$(arg robot)">
  <node pkg="people_detection_vision" type="tabletop_pplp.exe"
    name="tabletop_pplp"  output="screen"
    args="_max_dist_m:=5 _min_blob_size_pix:=3000 _max_blobs_nb:=1">
    <!-- autostart param - must be before the XML tag of the skill (node) -->
    <param if="$(arg autostart)" name="autostart" value="true"/>
  </node>

  <!-- conversion to ROS rviz marker -->
  <node if="$(arg marker)" pkg="vision_utils" type="ppl2marker.exe"
        output="screen" name="ppl2marker_nite_user_mask"
        args="_red:=1 _green:=0 _blue:=1 _input_topic:=tabletop_pplp/ppl" />

  <!-- event GUI -->
  <node if="$(arg eventgui)" pkg="ad_core" type="launcher_qt_event_sender_interface.exe"
        name="tabletop_pplp_event_gui"
        output="screen"
        args='event_keys="|TABLETOP_PPLP_START;TABLETOP_PPLP_START;0
                          |TABLETOP_PPLP_STOP;TABLETOP_PPLP_STOP;0"'/>
</group>
</launch>

